great_expectations/render/renderer/content_block/content_block.py

```diff
@@ -210,49 +117,177 @@ def render(cls, render_object, **kwargs):
+                    if isinstance(obj_, ExpectationValidationResult):
                         content_block_fn = cls._get_content_block_fn(
                             "_missing_content_block_fn"
                         )
+                        expectation_config = obj_.expectation_config
                         result = content_block_fn(
-                            result=render_object,
+                            configuration=expectation_config,
+                            result=obj_,
                             runtime_configuration=runtime_configuration,
                             **kwargs,
                         )
                     else:
                         content_block_fn = cls._missing_content_block_fn
                         result = content_block_fn(
-                            configuration=render_object,
+                            configuration=obj_,
                             runtime_configuration=runtime_configuration,
                             **kwargs,
                         )
             else:
-                if isinstance(render_object, ExpectationValidationResult):
+                if isinstance(obj_, ExpectationValidationResult):
                     content_block_fn = (
                         cls._get_content_block_fn("_missing_content_block_fn")
                         if not exception_list_content_block
                         else cls._missing_content_block_fn
                     )
+                    expectation_config = obj_.expectation_config
                     result = content_block_fn(
-                        result=render_object,
+                        configuration=expectation_config,
+                        result=obj_,
                         runtime_configuration=runtime_configuration,
                         **kwargs,
                     )
                 else:
                     content_block_fn = cls._missing_content_block_fn
                     result = content_block_fn(
-                        configuration=render_object,
+                        configuration=obj_,
                         runtime_configuration=runtime_configuration,
                         **kwargs,
                     )
+
             if result is not None:
-                if isinstance(render_object, ExpectationConfiguration):
-                    expectation_meta_notes = cls._render_expectation_meta_notes(
-                        render_object
-                    )
+                if isinstance(obj_, ExpectationConfiguration):
+                    expectation_meta_notes = cls._render_expectation_meta_notes(obj_)
                     if expectation_meta_notes:
-                        result.append(expectation_meta_notes)
-            return result
+                        # this adds collapse content block to expectation string
+                        result[0] = [result[0], expectation_meta_notes]
+
+                    horizontal_rule = RenderedStringTemplateContent(
```

Comment:
I feel like we lost the point of the type system here, how could we retain it?

Reply:
in regards to which line? 165 or 167?

Reply:
167, with the dictionary init vs using the types

Reply:
This is fine for now, actually